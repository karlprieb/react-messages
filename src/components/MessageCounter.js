import React from 'react'
import PropTypes from 'prop-types'
import { IconChat, IconMail } from '../icons'

import './messageCounter.css'

const MessageCounter = ({ counter }) => (
  <header className='message-counter'>
    <div className='message-counter-group'>
      <IconChat title='Chat bubles' />
      <span className='message-counter-text'>{counter} Online Requests</span>
    </div>
    <div className='message-counter-group'>
      <IconMail title='Envelope' />
      <span className='message-counter-text'>{counter} Offline Requests</span>
    </div>
  </header>
)

MessageCounter.propTypes = {
  counter: PropTypes.number.isRequired
}

export default MessageCounter
