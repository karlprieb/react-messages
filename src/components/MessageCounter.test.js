import React from 'react'
import MessageCounter from './MessageCounter'

describe('MessageCounter component', () => {
  it('should render app header with counter', () => {
    const wrapper = shallow(<MessageCounter counter={10} />)
    expect(wrapper).toMatchSnapshot()
  })
})
