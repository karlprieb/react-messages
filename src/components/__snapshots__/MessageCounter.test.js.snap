// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageCounter component should render app header with counter 1`] = `
<header
  className="message-counter"
>
  <div
    className="message-counter-group"
  >
    <IconChat
      title="Chat bubles"
    />
    <span
      className="message-counter-text"
    >
      10
       Online Requests
    </span>
  </div>
  <div
    className="message-counter-group"
  >
    <IconMail
      title="Envelope"
    />
    <span
      className="message-counter-text"
    >
      10
       Offline Requests
    </span>
  </div>
</header>
`;
